ui/ui.go

```diff
@@ -96,27 +96,41 @@ func (ui *WriterUI) PrintTable(table Table) {
 	}
 }
 
-func (ui *WriterUI) AskForText(label string) (string, error) {
+func (ui *WriterUI) AskForText(opts TextOpts) (string, error) {
 	var text string
 
-	err := interact.NewInteraction(label).Resolve(&text)
-	if err != nil {
-		return "", fmt.Errorf("Asking for text: %s", err)
+	if opts.ValidateFunc == nil {
+		opts.ValidateFunc = func(s string) (bool, string, error) {
+			return true, "", nil
+		}
 	}
 
-	return text, nil
+	for {
+		text = opts.Default
+		err := interact.NewInteraction(opts.Label).Resolve(&text)
+		if err != nil {
+			return "", fmt.Errorf("Asking for text: %s", err)
+		}
+
+		isValid, message, err := opts.ValidateFunc(text)
+		if err != nil {
+			return "", fmt.Errorf("Validation Error: %s", message)
```

Comment:
Here made changes such that `for` loop will continue only when input is invalid and err is nil. Let me know your thoughts.